pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1386,7 +1386,9 @@ func (c *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID str
 	}
 	if len(instances) == 0 {
 		glog.Warningf("the instance %s does not exist anymore", providerID)
-		return true, nil
+		// returns false, because otherwise node is not deleted from cluster
+		// false means that continue to to check InstanceExistsByProviderID
```

Comment:
Something is wrong with this sentence. *to* is twice and it feels like a word is missing between *that* and *continue* :)